server/src/main/java/org/elasticsearch/search/aggregations/bucket/filter/FiltersAggregator.java

```diff
@@ -289,6 +290,10 @@ static class Compatible extends FiltersAggregator {
 
         @Override
         protected LeafBucketCollector getLeafCollector(AggregationExecutionContext aggCtx, LeafBucketCollector sub) throws IOException {
+            if (filters().isEmpty() || (filters().size() == 1 && filters().get(0).query() instanceof MatchNoDocsQuery)) {
```

Comment:
Maybe we should check whether all filters are `MatchNoDocsQuery`?

Reply:
Wouldn't these get collapsed into a single MatchNoDocsQuery during merging/rewriting?

Reply:
Updated to play it safe.